### ISSUE TITLE
RHEL 9.1 error

### ISSUE THREAD

**Issue by srosciszewski** [eonpatapon/gnome-shell-extension-caffeine#218]
Technical details:

SyntaxError: expected property name, got '{'

Stack trace:
  _init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:216:33
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:128:33
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  start@resource:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17

**Comment by pakaoraki** [comment]
I can confirm the issue: the extension work as expected, but we get this error when opening preferences. RHEL 9.1 use gnome 40.4 and Caffeine v42.

**Comment by pakaoraki** [comment]
The problem come from this [commit](https://github.com/eonpatapon/gnome-shell-extension-caffeine/commit/26cbfb89aa910e32ec25e356c568259016840b7c#diff-c34f5a08bb4dc9f07bc07f46a87d92b7e7c17dd5521cd1b8709080d9950230a8). it seems like the `ShortcutSettingWidget` cause this error. I can't really find the root of the problem for the moment. Also, I reckon that using `Adw` in gnome 40 would generate another crash: `Error: Requiring Adw, version none: Typelib file for namespace 'Adw' (any version) not found.`

So maybe the best thing to do is simply removing this feature for gnome 40. So, create a specific fixed branch for gnome 40 and keep the current v42 for gnome 41/42. What do you think @eonpatapon ?

**Comment by pakaoraki** [comment]
Hi @srosciszewski 

I fix this issue and I test it successfully in RHEL 9.1 / gnome 40. Could you please confirm that it work for you ?

Remove your current version of Caffeine, then:
```
git clone --branch gnome-shell-40 https://github.com/pakaoraki/gnome-shell-extension-caffeine.git
cd gnome-shell-extension-caffeine
make build
make install
```

After logout, preferences should be accessible again.

**Comment by stuarthayhurst** [comment]
I'm inclined to close this for the same reasons as #256, unless @pakaoraki plans to merge a fix into the GNOME 40 branch. However, that won't make it to a full release, and users would have to be directed to that branch, but at least a fix would be available for users running old versions of GNOME.

**Comment by pakaoraki** [comment]
I guess there is not much people using Caffeine with Gnome 40, this fix was working for me but I don't have any feedback from users. Maybe we can still create a dedicated branch for gnome 40 with this fix in case some people want to use it ?

**Comment by stuarthayhurst** [comment]
There is a branch for GNOME 40 -> 42, under `gnome-shell-40-42`. I don't know what @eonpatapon's policy for updating older releases is, but perhaps your fix could be tested and merged into that branch, allowing this issue to be closed?

**Comment by stuarthayhurst** [comment]
Is it possible to disable the keyboard shortcut for just GNOME 40? That way the branch won't need to be modified.

Theoretically, you can publish an update that only lists GNOME 40 / 41 / 42, and the update will only reach those systems and leave GNOME 43 / 44 on the real latest version, but I haven't actually attempted that. I'll ask on GNOME's Matrix chatroom for extensions if they have any ideas.

**Comment by stuarthayhurst** [comment]
> Stuart
> >So I'm helping out with Caffeine, and we've got some reports the settings are broken on GNOME 40 from users with LTS distros. As the latest version doesn't support GNOME 40 anymore, can we push a new version that supports just GNOME 40 in the metadata with a fix, and resume normal releases for GNOME 43+ after that?

> Just Perfection
> >yes, multi versioning is supported on ego.
> >you can also inactivate the faulty version.

So that's an option then. I'm not sure inactivating the faulty version would work, as it would depend on when the issue was introduced, and when support was added for the other versions.

**Comment by eonpatapon** [comment]
I don't think it's necessary to deactivate the old version, as new install will use the latest one and probably existing installs will be prompted to update.

So if you have a patch, just push it to the branch and I'll upload a new version in e.g.o

**Comment by stuarthayhurst** [comment]
With GNOME 45 coming up, I don't think it's worth maintaining the extension for ancient versions of GNOME.

I'm happy to close this, unless someone really wants to get this fix pushed through to a release.

**Comment by pakaoraki** [comment]
> Is it possible to disable the keyboard shortcut for just GNOME 40? That way the branch won't need to be modified.

For some reason, I didn't manage to make a unique version that can be use in `Gnome 40` and `Gnome 41/42`. To make it work, the only solution I have is to publish a specific release for `Gnome 40` without the incompatible lines of code. 

I created a `gnome-shell-40` branch from : [gnome-shell-40](https://github.com/pakaoraki/gnome-shell-extension-caffeine/tree/gnome-shell-40). I agree that it's not worthy to maintain old Gnome version, but this way people having this issue can manually install the fix from this archived branch.

**Comment by pakaoraki** [comment]
Fixed in #293
